src/features/beacons/selectors.ts

```diff
@@ -24,15 +26,15 @@ export const getSelectedBeaconIds = selectionForSubset(globalIdToBeaconId);
  * Funciton that returns the name of the given beacon that should be shown to
  * the user on the UI.
  */
-export const getBeaconDisplayName = (beacon) =>
+export const getBeaconDisplayName = (beacon: Beacon) =>
   (beacon ? beacon.name || beacon.id : null) || 'Unnamed beacon';
 
 /**
  * Selector that returns the IDs of all the beacons that have no basic
  * information fetched from the server yet.
  */
 export const getBeaconIdsWithoutBasicInformation = createSelector(
-  (state) => state.beacons,
+  (state: RootState) => state.beacons,
   (beacons) => {
     const result = [];
 
```

Comment:
This really looks to me like it should just be a `filter` followed by a `map`:
```ts
export const getBeaconIdsWithoutBasicInformation: AppSelector<Identifier[]> =
  createSelector(getBeaconsAsCollection, (beacons) =>
    Object.entries(beacons.byId)
      .filter(([_key, value]) => value.name === undefined)
      .map(([key, _value]) => key)
  );
```

Reply:
Feel free to commit the improvement, but this comment is also not related to the changes in the PR.